lib/generators/rolemodel/simple_form/templates/app/inputs/switch_checkbox_input.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# SwitchCheckboxInput is a custom input type for SimpleForm that renders a checkbox as a switch control
+#
+# Options:
+#   :small - renders a smaller switch control
+#   :label_after_input - renders the label after the switch control
+#   :wrapper - the wrapper to use for this input, defaults to :inline_switch_wrapper. :switch_wrapper is also available.
+#
+# Usage:
+# <%= f.input :my_field, as: :switch_checkbox %>
+# <%= f.input :my_field, as: :switch_checkbox, small: true %>
+# <%= f.input :my_field, as: :switch_checkbox, label_after_input: true %>
+# <%= f.input :my_field, as: :switch_checkbox, wrapper: :switch_wrapper %>
+class SwitchCheckboxInput < SimpleForm::Inputs::BooleanInput
+  def input(wrapper_options = nil)
+    merged_input_options = merge_wrapper_options(input_html_options, wrapper_options)
+
+    switch_group = template.content_tag(:div, class: "switch #{'switch--small' if options[:small]}") do
+      build_check_box_without_hidden_field(merged_input_options) +
```

Comment:
Is this related to the `include_hidden: false` flag that the `check_box` form helper exposes.

Usually that hidden field is important, so just checking that the switch produces the same underlying form inputs that the form builder does.

Reply:
The html this generates looks like the following:

```html
<div class='form-group form-group--inline'>
  <div class="switch">
    <input type="checkbox" id="switch-demo" >
    <label for="switch-demo"></label>
  </div>
  <label class="form-label" for="switch-demo">Switch Text</label>
</div>
```

Since it already has a checkbox in there, it won't need the form builder hidden field

Reply:
I tested in a demo application and everything worked great!

Reply:
Just want to make sure we're speaking the same language.  Checkboxes which are "unchecked" do not submit their value to the server.  Since we often use them in Rails apps to represent a Boolean column, we also need a hidden input with the same name and the inverse value to come before the rendered checkbox in the form.  The hidden input is overwritten by the checkbox when it is is checked, because it has the same name.  When the checkbox is not checked, only the hidden input's value is submitted.  Without the hidden input, a checkbox can set a Boolean value to "true", but it cannot later update it back to "false".

Is this component meant for a different use-case than the one mentioned above?

Reply:
Ah yeah I didn't fully test it. You are correct.

I've updated the input helper. The outputted HTML would look like 

```
<div class="switch ">
  <input name="project[test]" type="hidden" value="0" autocomplete="off">
  <input type="checkbox" value="1" checked="checked" name="project[test]" id="project_test">
  <label class="switch_checkbox optional form-label" for="project_test">Test</label>
</div>
```

And it looks like 
<img width="363" alt="Screenshot 2023-08-11 at 11 14 00 AM" src="https://github.com/RoleModel/rolemodel_rails/assets/5957102/80c06094-2de3-407d-af93-f638ba7f0775">

(though this is dependent on the options passed)